gc.c

```diff
@@ -7021,7 +7023,9 @@ gc_aging(rb_objspace_t *objspace, VALUE obj)
     if (!RVALUE_PAGE_WB_UNPROTECTED(page, obj)) {
         if (!RVALUE_OLD_P(obj)) {
             gc_report(3, objspace, "gc_aging: YOUNG: %s\n", obj_info(obj));
-            RVALUE_AGE_INC(objspace, obj);
+            if (is_full_marking(objspace) || !gc_params.aging_factor || (objspace->profile.minor_gc_count % gc_params.aging_factor)) {
```

Comment:
Maybe we can use a flag `objspace->age_objects_p` which is set at the beginning of marking. This should improve performance because we won't need to use the modulo here.